src/main/java/com/metamx/common/parsers/TimestampParser.java

```diff
@@ -60,46 +60,28 @@ public DateTime apply(String input)
           return new DateTime(ParserUtils.stripQuotes(input));
         }
       };
-    } else if (format.equalsIgnoreCase("posix")) {
+    } else if (format.equalsIgnoreCase("posix")
+               || format.equalsIgnoreCase("millis")
+               || format.equalsIgnoreCase("nano")) {
+      final Function<Number, DateTime> numericFun = createNumericTimestampParser(format);
       return new Function<String, DateTime>()
       {
         @Override
         public DateTime apply(String input)
         {
           Preconditions.checkArgument(input != null && !input.isEmpty(), "null timestamp");
-          return new DateTime(Long.parseLong(ParserUtils.stripQuotes(input)) * 1000);
+          return numericFun.apply(Long.parseLong(ParserUtils.stripQuotes(input)));
         }
       };
     } else if (format.equalsIgnoreCase("ruby")) {
```

Comment:
TimestampParserTest.testRuby verifies ruby implementation. 
Added comments in code.